### PR TITLE
Add integration test for /getallchats API in getallchatsTestcase branch

### PR DESCRIPTION
**Pull request by santhiya-mj**
Implement integration testing for the /getallchats API endpoint to verify retrieval of all chats, updating changes to the getallchatsTestcase branch.

### CODE REVIEW

File: test/IntegrationTests/test_getallchats.py [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+            "chat_id": self.valid_chat_id
+        }
+        logger.info(f"INPUT: Sending request with data:\n{str(data)}")
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        self.assertEqual(response.status_code, 404)
+        logger.info("=== Test Case 3 Completed ===\n")
```
Comment (SaravanakumarR2018) [inline test/IntegrationTests/test_getallchats.py:102]: @santhiya-mj you should verify the response for the text as well
self.assertIn("no chats found for this customer and chat id", response.text.lower())

File: test/IntegrationTests/test_getallchats.py [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        # Verify that the response status code is 200
+        self.assertEqual(response.status_code, 200, "Expected status code 200")
+
+        messages = response.json().get('messages')
+        self.assertIsInstance(messages, list)  # Check that messages is a list
+
+        # Verify that the chat_id and customer_guid in the response match the input
+        if messages:  # Check if messages list is not empty
```
Comment (SaravanakumarR2018) [inline test/IntegrationTests/test_getallchats.py:128]: First you should add chats to this customer_guid and chat_id and then verify whether each of the added chat messages have arrived back 

The chat messages added should exactly come as output without missing a single chat message

File: test/IntegrationTests/test_getallchats.py [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+        logger.info("=== Starting Test Case 6: Valid customer_guid, chat_id, page=1, and page_size=1 ===")
+
+        url = f"{self.BASE_URL}/getallchats"
+        data = {
+            "customer_guid": self.valid_customer_guid,
+            "chat_id": self.valid_chat_id,
+            "page": 1,
+            "page_size": 1
+        }
+        logger.info(f"INPUT: Sending request with data:\n{str(data)}")
```
Comment (SaravanakumarR2018) [inline test/IntegrationTests/test_getallchats.py:184]: First we should add a chat message and then verify retrieving it with page as 1 and page_size as 1 and verify the one retrieved should exactly match the input

File: test/IntegrationTests/test_getallchats.py [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+
+        self.assertEqual(response.status_code, 404)
+        logger.info("=== Test Case 10 Completed ===\n")
```
Comment (SaravanakumarR2018) [inline test/IntegrationTests/test_getallchats.py:270]: Can we also verify the output text here

File: test/IntegrationTests/test_getallchats.py [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+        }
+        logger.info(f"INPUT: Sending request with data:\n{str(data)}")
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        response_data = response.json()
+
+        # Adjusted to check that 'messages' is a list
```
Comment (SaravanakumarR2018) [inline test/IntegrationTests/test_getallchats.py:291]: You should first add messages and then verify whether there are arriving

File: test/IntegrationTests/test_getallchats.py
Comment (SaravanakumarR2018) [inline test/IntegrationTests/test_getallchats.py:1]: @santhiya-mj 

Can we add two more testcases

1) Add a testcase with 5 chats, then retrieve only 2 chats and verify whether only 2 chats are arriving using page 1 and page size 2, then retrieve the next chats by increasing page number until all the chats are exhausted and verify all the chats are arriving
2) Add a testcase with 5 chats, then retrieve 20 chats  and verify all the sent chats are arriving

File: test/IntegrationTests/test_getallchats.py (hunk 1 above)
Comment (santhiya-mj) [reply test/IntegrationTests/test_getallchats.py:107]: Done sir, I've added the assertion to verify the response text.

File: test/IntegrationTests/test_getallchats.py (hunk 2 above)
Comment (santhiya-mj) [reply test/IntegrationTests/test_getallchats.py:164]: Done sir, all chat messages retrieved.

File: test/IntegrationTests/test_getallchats.py (hunk 3 above)
Comment (santhiya-mj) [reply test/IntegrationTests/test_getallchats.py:244]: Done sir, chat message added and verified with exact match.

File: test/IntegrationTests/test_getallchats.py (hunk 4 above)
Comment (santhiya-mj) [reply test/IntegrationTests/test_getallchats.py:360]: Done sir, I've added the assertion to verify the response text.

File: test/IntegrationTests/test_getallchats.py (hunk 5 above)
Comment (santhiya-mj) [reply test/IntegrationTests/test_getallchats.py]: Done sir, messages added and verified successfully.

**Pull request closed without merging**